File: webscripts/shared/src/main/java/com/github/dynamicextensionsalfresco/webscripts/AnnotationWebScriptRequest.java
package com.github.dynamicextensionsalfresco.webscripts;

import java.util.LinkedHashMap;
import java.util.Map;
import org.springframework.extensions.webscripts.WebScriptRequest;
import org.springframework.extensions.webscripts.WrappingWebScriptRequest;

public class AnnotationWebScriptRequest extends DelegatingWebScriptRequest implements WrappingWebScriptRequest {

    private final Map<String, Object> model = new LinkedHashMap<>();
    private Throwable thrownException = null;

    public AnnotationWebScriptRequest(WebScriptRequest request) {
        super(request);

    }

    public WebScriptRequest getWebScriptRequest() {
        return this.getNext();
    }


    public Map<String, Object> getModel() {
        return model;
    }

    public Throwable getThrownException() {
        return thrownException;
    }

    public void setThrownException(Throwable thrownException) {
        this.thrownException = thrownException;
    }

}

File: webscripts/src/main/kotlin/com/github/dynamicextensionsalfresco/webscripts/AnnotationWebScriptRequest.kt
package com.github.dynamicextensionsalfresco.webscripts

import org.springframework.extensions.webscripts.WebScriptRequest
import org.springframework.extensions.webscripts.WrappingWebScriptRequest
import java.util.LinkedHashMap

public class AnnotationWebScriptRequest(val webScriptRequest: WebScriptRequest) : WebScriptRequest by webScriptRequest, WrappingWebScriptRequest {
    override fun toString(): String {
        return next.toString()
    }

    public val model: MutableMap<String, Any> = LinkedHashMap()

    public var thrownException: Throwable? = null
        set

    override fun getNext(): WebScriptRequest {
        if (webScriptRequest is WrappingWebScriptRequest) {
            return webScriptRequest.next
        } else {
            return webScriptRequest
        }
    }
}
